schematic_api/api/routes.py

```diff
@@ -206,130 +220,193 @@ def get_temp_jsonld(schema_url):
-                else: 
-                    t = title
-            if dataset_ids:
-                # if a dataset_id is provided add this to the function call.
-                result = create_single_manifest(data_type=dt, dataset_id=dataset_ids[i], output_format=output_format, title=t, access_token=access_token)
-            else:
-                result = create_single_manifest(data_type=dt, output_format=output_format, title=t, access_token=access_token)
-
-            # if output is pandas dataframe or google sheet url
-            if isinstance(result, str) or isinstance(result, pd.DataFrame):
-                all_results.append(result)
-            else: 
-                if len(data_type) > 1:
-                    app.logger.warning(f'Currently we do not support returning multiple files as Excel format at once. Only {t} would get returned. ')
-                return result
-
-    return all_results
+    @staticmethod
+    def get_manifests_route(schema_url: str, data_type:List[str], use_annotations: Optional[str]=None, dataset_id: Optional[List[str]]=None, title: Optional[str]=None, asset_view: Optional[str]=None, output_format: Optional[str]="google_sheet") -> str|pd.DataFrame|BinaryIO:
+        """Generate a new manifest template or create an existing manifest in google sheet/excel/dataframe format. 
+
+        Args:
+            schema_url (str): data model in json ld format.
+            output_format (str, optional): output format of manifests. Available options are: google sheet, excel, and dataframe. Defaults to "google_sheet".
+            use_annotations (bool, optional): whether to use existing annotations during manifest generation. Defaults to False
+            dataset_id (list[str], optional): id of a given dataset. Defaults to None.
+            data_type (list[str], optional):a list of data types. Defaults to None.
+            asset_view (str, optional): id of a file view. Defaults to None.
+            title (str, optional): title of the new manifest. Defaults to None.
+
+        Raises:
+            ValueError: could not submit data_type = "all manifests" and dataset id at the same time
+            ValueError: do not enter multiple dataset ids if calling "all manifests" for data_type.
+
+        Returns:
+            str|pd.DataFrame|BinaryIO: returns google sheet url, pandas dataframe, or excel file based on output_format parameter
+        """
+        # initalize manifest generation class
+        access_token=request.headers.get('X-Auth')
+        mg = ManifestGeneration(schema_url=schema_url, access_token=access_token, output_format=output_format, title=title, use_annotations=use_annotations, dataset_id=dataset_id, data_type=data_type, asset_view=asset_view)
+
+        # load configuration file
+        mg._load_config_(app)
+
+        # get path to temporary JSON-LD file
+        jsonld = get_temp_jsonld(schema_url)
+
+        # Gather all returned result urls
+        if data_type[0] == 'all manifests':
```

Comment:
This looks like a magic (_i.e._ hard-coded) value. I wonder if there's another way to achieve this behavior that's more explicit.